### PR TITLE
feat: EMA slice 3 — org onboarding UI, resource-cred fields, expiry banner (END-19)

### PR DESCRIPTION
**Pull request by panghy**
## EMA slice 3 — org onboarding UI, resource-cred fields, expiry banner (END-19)

Desktop-side slice-3 refinements for the Enterprise-Managed Authentication (EMA) chain landed in the END-19 base PRs (relay#121 / desktop#140). Pairs with [`endara-ai/endara-relay#127`](https://github.com/endara-ai/endara-relay/pull/127).

### Highlights

- **Org onboarding UI** — `ConnectOrgModal` and `OrganizationsSection` pick up the org lifecycle updates from relay (confidential-client secret, resource-scope defaults) and expose a full org edit path via the new `EditOrganizationModal`.
- **Add Server — resource-credential fields** — `AddEndpointModal` gains per-endpoint `resource_client_id`, `resource_client_secret`, and `resource_scope` inputs in a consolidated **Advanced** section. Values flow through `add-endpoint-helpers.ts` and the Tauri command surface into the endpoint's own `{name}.dcr.json` store server-side, so credentials never leak onto the IdP-facing legs.
- **Consolidated Advanced section** — the previously scattered advanced fields collapse into a single expandable section in the Add Server modal, keeping the default flow simple.
- **Re-auth polling** — `onboarding-helpers.ts` drives a poll that flips the Settings row to **Connected** once the resource legs succeed, so the user does not need to reload after finishing SSO.
- **Full-width org-expiry banner** — new `OrgExpiryBanner` component wired into `+layout.svelte` renders app-wide above the main content whenever an org's tokens are near expiry, with a 30-second poll that refreshes org state so the banner appears/disappears without user interaction (and so the Settings pane's per-org rows stay in sync).

### Commits

- `feat: EMA slice 3 — org onboarding UI, per-endpoint resource-cred fields, re-auth polling (END-19)`
- `feat: full-width org-expiry banner + app-wide org poll`

### Verification

- `pnpm check` — 0 errors / 0 warnings
- `pnpm test` — 1081 passed / 39 skipped / 0 failed
- `pnpm build` — clean

Refs END-19.

### CODE REVIEW

**Review by panghy — COMMENTED** [review]
### Review — END-19 EMA slice 3 (org onboarding UI, resource-cred fields, expiry banner)

Overall this looks solid. The absent-vs-empty merge semantics for the resource pair, the write-only secret handling (nothing echoed back or logged), the shared `pollForOrgAuth` extraction with proper cancellation, and the overlay-window guard (poll early-returns in `onMount` and the banner is gated by `{#if !isOverlayWindow}`) are all correct and well covered by tests. One actionable concern plus one minor edge case:

---

**1. (main) Full-width `OrgExpiryBanner` overlaps the top tab bar / window drag region — `OrgExpiryBanner.svelte` (the `role="alert"` div) + `+layout.svelte:67-69`**

The banner is `position: fixed; top-0; left-0; right-0; z-50`, but the app root (`+page.svelte:256`) is a `flex flex-col h-screen w-screen overflow-hidden` starting at `top:0`, whose first child is the top-level tab bar (`+page.svelte:258`) that also carries `data-tauri-drag-region` (and on macOS sits under the traffic-light buttons).

Because the banner is `fixed` and nothing adds a compensating `padding-top`/spacer to `{@render children()}`, whenever an org token expires the banner renders **on top of** the tab bar — obscuring the Servers/Catalog/Profiles/Observability/Logs/Settings tabs and covering the draggable title strip / traffic-light area.

*Why it matters:* the primary top-level navigation becomes unclickable while the banner is up, and the frameless-window drag region is blocked.

*Suggested fix:* reserve space when the banner is visible rather than overlaying — e.g. make the banner participate in layout (render it as the first flex child of the app column instead of `fixed`), or add a matching `padding-top`/spacer on the content container that is applied only while `orgExpiryBannerMessage(...)` is non-null. If overlaying the tab bar is intentional, please confirm the drag region / traffic-lights remain usable.

---

**2. (minor) `EditOrganizationModal.svelte` — switching between two *templated* providers with a blank slug sends `provider` without `slug`**

`handleSave` only enforces a required issuer/slug when the *new* provider is a custom provider (`isCustomProvider(selectedProvider) && providerId !== org.provider`). If the user changes provider between two templated providers (e.g. `okta` → `ping`) and leaves the slug blank ("leave blank to keep current"), `buildParams()` emits `{ provider: 'ping' }` with no `slug`. "Keep current" issuer is meaningless across a template change, so the relay has nothing to rebuild the new issuer from.

*Suggested fix:* when `providerId !== org.provider` and the new provider `providerNeedsSlug(...)`, require a non-empty slug (mirror the existing custom-provider guard).

---

Nothing here is release-blocking on its own; #1 is the one I'd want addressed (or explicitly confirmed as intended) before merge.

File: src/lib/api.ts [hunk 1]
```diff
@@ -295,6 +345,14 @@ export interface EndpointConfig {
...
   mounts?: string[];
+  /**
+   * EMA org-binding mirrored from the persisted `[endpoints.auth]` sub-table.
+   * Absent for ordinary endpoints. The UI displays the bound organization
+   * (read-only) and passes the block back on `updateEndpoint` so the relay's
+   * PUT — which rebuilds the whole endpoint config from the body — preserves
+   * the binding.
+   */
+  auth?: EmaAuthConfig;
 }
```
[the remark this answers is not in the document]
Comment (panghy) [reply src/lib/api.ts:356]: Acknowledged — valid type-precision nit. Reusing EmaAuthConfig for the read/round-trip side is intentional for this slice; introducing a dedicated read-side summary type with optional organization/resource fields is tracked as a follow-up refinement and is not blocking END-19 slice 3. Resolving as a deferred, non-blocking nit.

File: src/lib/api.ts [hunk 2]
```diff
@@ -342,6 +411,14 @@ export interface UpdateEndpointParams {
...
    */
   mounts?: string[];
+  /**
+   * EMA org-binding block (`[endpoints.auth]`) round-tripped on update. The
+   * relay's PUT rebuilds the whole endpoint config from the body, so the
+   * stored binding must be sent back verbatim or it would be silently dropped.
+   * Absent for ordinary (non-EMA) endpoints — no empty `auth` key is sent in
+   * that case, keeping the PUT body byte-for-byte unchanged.
+   */
+  auth?: EmaAuthConfig;
```
[the remark this answers is not in the document]
Comment (panghy) [reply src/lib/api.ts:421]: Acknowledged — same type-precision nit as on EndpointConfig.auth. UpdateEndpointParams.auth intentionally reuses EmaAuthConfig for this slice; a dedicated summary type with optional fields is a follow-up refinement, not blocking END-19 slice 3. Resolving as a deferred, non-blocking nit.

File: src/lib/components/ConnectOrgModal.svelte [hunk 3]
```diff
@@ -42,6 +43,8 @@
   let orgName = $state('');
   let slugOrUrl = $state('');
   let clientId = $state('');
+  // Write-only; never echoed back. Cleared on cancel/submit failure.
+  let clientSecret = $state('');
```
[the remark this answers is not in the document]
Comment (panghy) [reply src/lib/components/ConnectOrgModal.svelte:47]: Acknowledged — the doc comment overstates the clearing behavior for clientSecret. This is a comment-accuracy nit only, with no functional impact; the wording will be tightened in a follow-up. Resolving as a non-blocking nit.
